query.go

```diff
@@ -26,20 +31,10 @@ func (q *Query) Exec() (Result, error) {
 	return Result(res), err
 }
 
-func (q *Query) pickConn() (*transport.Conn, error) {
-	token, tokenAware := q.token()
-	info, err := q.info(token, tokenAware)
-	if err != nil {
-		return nil, err
-	}
-	n := q.session.cfg.HostSelectionPolicy.Node(info, 0)
+func (q *Query) pickConn(qi transport.QueryInfo) (*transport.Conn, error) {
```

Comment:
Nit: Rewrite commit message "session: pickConn, take QueryInfo as argument instead of calculating it" to something better like: "session: take QueryInfo in pickConn and node.Conn()".